tests/e2e/src/test/java/org/glassfish/jersey/tests/api/ContentDispositionTest.java

```diff
@@ -97,6 +97,136 @@ public void testToString() {
+            fileNameExt = "ISO-8859-1'language-us'a%";
+            header = prefixHeader + fileNameExt + "\"";
+            contentDisposition = new ContentDisposition(HttpHeaderReader.newInstance(header), true);
+            assertEquals(fileName, contentDisposition.getFileName());
+
+            //correct fileNameExt
+            fileNameExt = "ISO-8859-1'language-us'abc%a1abc%a2%b1!#$&+.^_`|~-";
+            header = prefixHeader + fileNameExt + "\"";
+            contentDisposition = new ContentDisposition(HttpHeaderReader.newInstance(header), true);
+            assertEquals(fileNameExt, contentDisposition.getFileName());
+
+            //correct fileNameExt
+            fileNameExt = "ISO-8859-1'language-us'abc%a1abc%a2%b1!#$&+.^_`|~-";
+            header = prefixHeader + fileNameExt + "\"";
+            contentDisposition = new ContentDisposition(HttpHeaderReader.newInstance(header), true);
+            assertEquals(fileNameExt, contentDisposition.getFileName());
+
+            //correct fileNameExt
+            fileNameExt = "iso-8859-1'language-us'abc.TXT";
+            header = prefixHeader + fileNameExt + "\"";
+            contentDisposition = new ContentDisposition(HttpHeaderReader.newInstance(header), true);
+            assertEquals(fileNameExt, contentDisposition.getFileName());
+
+            //incorrect fileNameExt
+            fileNameExt = "ISO-8859-1'language-us'";
+            header = prefixHeader + fileNameExt + "\"";
+            contentDisposition = new ContentDisposition(HttpHeaderReader.newInstance(header), true);
+            assertEquals(fileName, contentDisposition.getFileName());
+
+            //incorrect fileNameExt
+            fileNameExt = "ISO-8859-1'language-us'c:\\file.txt";
+            header = prefixHeader + fileNameExt + "\"";
+            contentDisposition = new ContentDisposition(HttpHeaderReader.newInstance(header), true);
+            assertEquals(fileName, contentDisposition.getFileName());
+
+            //incorrect fileNameExt
+            fileNameExt = "ISO-8859-1'language-us'home/file.txt";
+            header = prefixHeader + fileNameExt + "\"";
+            contentDisposition = new ContentDisposition(HttpHeaderReader.newInstance(header), true);
+            assertEquals(fileName, contentDisposition.getFileName());
+
+            //incorrect fileNameExt
+            fileNameExt = "ISO-8859-1'language-us'李.txt";
+            header = prefixHeader + fileNameExt + "\"";
+            contentDisposition = new ContentDisposition(HttpHeaderReader.newInstance(header), true);
+            assertEquals(fileName, contentDisposition.getFileName());
+
+            //correct fileNameExt
+            fileNameExt = "ISO-8859-1'language-us'FILEname.tXt";
+            header = prefixHeader + fileNameExt + "\"";
+            contentDisposition = new ContentDisposition(HttpHeaderReader.newInstance(header), true);
+            assertEquals(fileNameExt, contentDisposition.getFileName());
```

Comment:
regarding those tests - it would be good to extract duplicated code into some method and call it only with those parameters which differs.

Reply:
fixed